pkg/scheduler/core/extender.go

```diff
@@ -83,6 +86,16 @@ func NewHTTPExtender(config *schedulerapi.ExtenderConfig) (algorithm.SchedulerEx
 		Transport: transport,
 		Timeout:   config.HTTPTimeout,
 	}
+	if config.Selector == nil {
```

Comment:
You should place `selector = labels.Everything()` inside this `if` and remove the other `if` below.

Reply:
Updated

```
selector := labels.Everything()
if config.Selector != nil {
	selector, err = metav1.LabelSelectorAsSelector(config.Selector)
	if err != nil {
		return nil, err
	}
}
```